### PR TITLE
Update: Add exceptionPatterns to id-length rule (fixes #13094)

### LINKED ISSUE #13094: id-length exceptions pattern
**What rule do you want to change?**
Property `exceptions` for rule `id-length`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Property value support

**Please provide some example code that this change will affect:**

```js
{
  id-length: {
    min: 3,
    max: 16,
    exceptions: ["^BEFORE_"]
  }
}
```

**What does the rule currently do for this code?**
It makes the exception only for identifiers named exactly `^BEFORE_`

**What will the rule do after it's changed?**
It will make the exception for all identifiers matching the pattern. In the example above, it will make the exception for every identifier starting with `BEFORE`

**Are you willing to submit a pull request to implement this change?**
Not for the moment

### PR DESCRIPTION
**Pull request by burawi**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

https://github.com/eslint/eslint/issues/13094

#### What changes did you make? (Give an overview)
I added an option named `exceptionPatterns` to `id-length`, which takes as value an array of the allowed patterns.
If the option is not defined it takes an empty array as default value.
Then I declared a boolean `matchesExceptions` that gets `true` if one of the patterns tests the name.
And added that boolean to the condition.

I wrote 2 tests:
- A valid one that uses `exceptionPatterns` option to allow the long identifier name.
- And another one, that would be invalid in which I put the same data as the previous test but this time without the new option.

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Marking as accepted since the issue is accepted now.

File: tests/lib/rules/id-length.js [hunk 1]
```diff
@@ -77,7 +77,8 @@ ruleTester.run("id-length", rule, {
         { code: "var {x} = foo;", options: [{ properties: "never" }], parserOptions: { ecmaVersion: 6 } },
         { code: "var {x, y: {z}} = foo;", options: [{ properties: "never" }], parserOptions: { ecmaVersion: 6 } },
         { code: "let foo = { [a]: 1 };", options: [{ properties: "always" }], parserOptions: { ecmaVersion: 6 } },
-        { code: "let foo = { [a + b]: 1 };", options: [{ properties: "always" }], parserOptions: { ecmaVersion: 6 } }
+        { code: "let foo = { [a + b]: 1 };", options: [{ properties: "always" }], parserOptions: { ecmaVersion: 6 } },
+        { code: "function BEFORE_send() {};", options: [{ min: 3, max: 5, exceptionPatterns: ["^BEFORE_"] }], parserOptions: { ecmaVersion: 6 } }
     ],
```
Comment (mdjermanovic) [inline tests/lib/rules/id-length.js:82]: Can we also add a `valid` test with multiple exception patterns?

**Comment by kaicataldo** [comment]
@burawi Are you still interested in working on this?

**Comment by burawi** [comment]
> @burawi Are you still interested in working on this?

I'm sorry, I'm a little bit busy currently I can't invest time in this

**Comment by aladdin-add** [comment]
Closing because there hasn't been activity for 30 days. If you're still interested in submitting this code, please feel free to resubmit.

**Pull request closed without merging**